packages/experience/src/Providers/ConfirmModalProvider/index.tsx

```diff
@@ -44,70 +42,68 @@ const defaultModalState: ConfirmModalState = {
 
 const ConfirmModalProvider = ({ children }: Props) => {
   const [modalState, setModalState] = useState<ConfirmModalState>(defaultModalState);
-
-  const resolver = useRef<(value: [result: boolean, data?: unknown]) => void>();
+  const [isConfirmLoading, setIsConfirmLoading] = useState(false);
+  const resolver = useRef<Resolver>();
```

Comment:
The point of using a resolver here is to make the confirmModal used a promise. If we go with the normal onConfirm/onCancell callback. This whole promisify logic is useless?

Reply:
updated.